papers/vyas_ramasubramani/94_vyas_ramasubramani.rst

```diff
@@ -348,28 +341,33 @@ As a result, our initial comparisons will focus on existing packages that solve
 
-.. TODO any plans to integrate the configuration/use of Sacred into signac? It sounds like any signac user could be encouraged to improve his/her provenance by supporting Sacred in signac? ... Well, nvm, you answered it in the next paragraph :)
+We have found that integrating Sacred with ``signac`` is in fact quite simple.
+Once functions are registered with either a Sacred Experiment or a ``signac-flow`` ``FlowProject``, the operations can be run either through Python or on the command line.
+While both tools typically advocate using their command line interfaces, the two can be integrated by using one from the command line while having it internally call the other through the other's Python interface.
+Since ``signac-flow`` provides a more extensive set of command line options that are likely to be regularly changed, we advocate using the ``signac-flow`` command line functionality and encoding a Sacred Experiment within a ``signac-flow`` operation.
+
+The Sumatra provenance tracking tool is an alternative to Sacred.
+Although it is written in Python, it is primarily designed for use as a command line utility, making it more suitable than Sacred for non-Python application.
+However, it does provide a Python API that offers greater flexibility than the command line tool, and this is the recommended mode for integration with ``signac-flow`` operations.
 
-However, Sacred is currently incompatible with ``signac-flow``.
-Sacred and ``signac-flow`` both depend on the registration of particular functions with some internal API: in ``signac-flow``, functions are stored as operations within the FlowProject, whereas Sacred tracks functions through the *Experiment* class.
-Since the actual script can only be run through one of these interfaces (whether on the command line or directly in Python), and while it is possible to use ``signac``'s database facilities in concert with Sacred, running operations using ``signac-flow`` prevents tracking them using Sacred.
-Conversely, the Sumatra provenance tracking tool *can* be integrated with ``signac``.
-Sumatra provides a command line utility for simple usage, but it also allows direct integration into Python scripts via a Python API, and it tracks similar information to Sacred.
-While the command line API is not flexible enough to allow passing arguments through to ``signac-flow``, the Python API can be easily integrated into ``signac-flow`` operations for tracking workflows managed by ``signac-flow``.
+We are currently assessing the explicit integration of one or both of these tools with ``signac-flow``.
+At present, the ``signac`` documentation contains example of how to integrate ``signac-flow`` with Sacred, and we plan to flesh out such examples further in the near future.
+Provenance tracking is a planned feature for the next release of ``signac-flow``, so that version will either contain its own implementation of provenance tracking or interface directly with one of these tools.
```

Comment:
Would it maybe make sense to be less detailed here? It is likely that this integration will have been worked out by the time that we actually present the paper at the conference in July.

Whatever solution we will have found out integration/no-integration, the paper would already be outdated by then.

Reply:
👍